### PR TITLE
fix: add PhysicalAddress migrations to postgres

### PR DESCRIPTION
**Pull request by jcmelati**

### CODE REVIEW

**Review by BtencateSphereon — CHANGES_REQUESTED** [review]
I don't think we released any products yet that use the migrations to production, so please update the existing migration to include the table. This so we keep the complexity down.

Besides that, please look at the naming conventions we use. and use this PR to sync the changes made for sqlite for postgress so we keep them the same.

https://github.com/Sphereon-Opensource/SSI-SDK/pull/161

**Comment by BtencateSphereon** [comment]
we do have the mobile wallet in the review process, but since this is using sqlite, i do not think this will impact the needed change for postgres. So i would say it is still safe to update the latest postgres migration, instead of making a new migration file. But we need to double check with @nklomp

**Comment by nklomp** [comment]
Yeah I am still okay with updating it for sqlite at this point. We are at 0.2 and moved to a completely different kv store forcing users to reonboard anyway. Once we are getting closer to 1.0 we will change of course to have new migration files for every single change

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 1]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
       `INSERT INTO "BaseContact"(id, legal_name, display_name, party_id, created_at, last_updated_at, type) SELECT id, name, alias, id, created_at, last_updated_at, 'Organization' FROM "Contact"`,
     )
     await queryRunner.query(`DROP TABLE "Contact"`)
+    
+    await queryRunner.query(`
+      CREATE TABLE "PhysicalAddress" (
```
Comment (BtencateSphereon) [inline packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts:90]: is missing a PRIMARY KEY constraint

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 2]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
...
+          "id" text PRIMARY KEY NOT NULL,
+          "type" varchar(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts:92]: as per documentation, `varchar` and `character varying(255)` are exactly the same thing
https://www.postgresql.org/docs/current/datatype-character.html

[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: sure, but we use character varying everywhere, so for consistency use that

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 3]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
...
+          "street_name" varchar(255) NOT NULL,
+          "street_number" varchar(255) NOT NULL,
+          "postal_code" varchar(255) NOT NULL,
+          "city_name" varchar(255) NOT NULL,
+          "province_name" varchar(255) NOT NULL,
+          "country_code" varchar(2) NOT NULL,
+          "building_name" varchar(255),
+          "partyId" uuid,
+          "created_at" timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Adjusted as per the code standard.

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 4]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
...
+          "last_updated_at" timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Adjusted as per the codebase standard.

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 5]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
       `INSERT INTO "BaseContact"(id, legal_name, display_name, party_id, created_at, last_updated_at, type) SELECT id, name, alias, id, created_at, last_updated_at, 'Organization' FROM "Contact"`,
     )
     await queryRunner.query(`DROP TABLE "Contact"`)
+    
+    await queryRunner.query(`
+      CREATE TABLE "PhysicalAddress" (
+          "id" text PRIMARY KEY NOT NULL,
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Fixed.

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Adjusted as per the indicated code base standard.

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 6]
```diff
@@ -85,6 +85,24 @@ export class CreateContacts1690925872592 implements MigrationInterface {
...
+          "street_number" varchar(255) NOT NULL,
+          "postal_code" varchar(255) NOT NULL,
+          "city_name" varchar(255) NOT NULL,
+          "province_name" varchar(255) NOT NULL,
+          "country_code" varchar(2) NOT NULL,
+          "building_name" varchar(255),
+          "partyId" uuid,
+          "created_at" timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+          "last_updated_at" timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+          CONSTRAINT "FK_PhysicalAddressEntity_partyId" FOREIGN KEY ("partyId") REFERENCES "Party" ("id") ON DELETE CASCADE
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Fixed.

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts (hunk 1 above)
Comment (jcmelati) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts:90]: Fixed.

**Review by BtencateSphereon — APPROVED** [review]

**Pull request merged**